readthedocs/projects/tasks/builds.py

```diff
@@ -554,6 +555,16 @@ def get_valid_artifact_types(self):
                 version=self.data.version.slug,
                 type_=artifact_type,
             )
+
+            if artifact_type == "html":
+                index_html_filepath = os.path.join(artifact_directory, "index.html")
+                if not os.path.exists(index_html_filepath):
+                    log.info(
+                        "Failing the build. "
+                        "HTML output does not contain an 'index.html' at its root directory.",
+                    )
```

Comment:
I believe we have support for serving `README.html` as the index file of docs projects. Does this logic impact those projects? I think that's the only place I could see this breaking in a way users yell at us.

Reply:
Are we OK checking by `index.html OR README.html` here?

Reply:
That is a solution that should work for now. We should deprecate that functionality as well, but it should go through the standard process, and needs to get in line 🤣